lib/benchee/benchmark/runner.ex

```diff
@@ -38,36 +50,24 @@ defmodule Benchee.Benchmark.Runner do
        ) do
     printer.benchmarking(job_name, input_name, config)
 
-    measurements =
-      1..config.parallel
-      |> Parallel.map(fn _ -> measure_scenario(scenario, scenario_context) end)
-      |> List.flatten()
+    config
+    |> measure_scenario_parallel(scenario, scenario_context)
+    |> add_measurements_to_scenario(scenario)
+  end
+
+  defp measure_scenario_parallel(config, scenario, scenario_context) do
+    1..config.parallel
+    |> Parallel.map(fn _ -> measure_scenario(scenario, scenario_context) end)
+    |> List.flatten()
+  end
 
+  defp add_measurements_to_scenario(measurements, scenario) do
     run_times = Enum.flat_map(measurements, fn {run_times, _} -> run_times end)
     memory_usages = Enum.flat_map(measurements, fn {_, memory_usages} -> memory_usages end)
 
-    memory_usages =
-      if Enum.all?(memory_usages, &is_nil/1) do
-        []
-      else
-        memory_usages
-      end
```

Comment:
I changed this at first but moved this decision to an earlier point now (see below)